src/vos/vos_gc.c

```diff
@@ -0,0 +1,838 @@
+/**
+ * (C) Copyright 2019 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+/**
+ * This file is part of daos
+ *
+ * vos/vos_gc.c
+ */
+#define D_LOGFAC	DD_FAC(vos)
+
+#include <daos/btree.h>
+#include <daos/mem.h>
+#include <daos_srv/vos.h>
+#include "vos_internal.h"
+
+enum {
+	GC_CREDS_MIN	= 1,	/**< minimum credits for vos_gc_run */
+	GC_CREDS_PRIV	= 128,	/**< credits for internal usage */
+	GC_CREDS_MAX	= 4096,	/**< maximum credits for vos_gc_run */
```

Comment:
These values looks bit small to me, PMDK support million ops in single transaction, larger credit could be good for efficiency.

Reply:
I'm worrying about regular I/O latency if they are too large, let's tune these values in real environment.

Reply:
I've changed it to 256, but I'd like to be cautious before we bump it again.